packages/core/src/utils/security/index.ts

```diff
@@ -91,12 +92,20 @@ function getAuthResults(securitySchemes: SecurityScheme[][], someInput: unknown,
   });
 }
 
-export function validateSecurity(someInput: unknown, resource?: unknown): Option<IPrismDiagnostic> {
+export function validateSecurity(
+  someInput: unknown,
+  resource?: unknown
+): Either.Either<NonEmptyArray<IPrismDiagnostic>, unknown> {
   const securitySchemes = get(resource, 'security', []);
 
   if (!securitySchemes.length) {
-    return none;
+    return Either.right(someInput);
   } else {
-    return gatherValidationResults(securitySchemes, someInput, resource);
+    return pipe(
+      gatherValidationResults(securitySchemes, someInput, resource),
+      Either.fromOption(() => someInput),
+      Either.swap,
+      Either.mapLeft<IPrismDiagnostic, NonEmptyArray<IPrismDiagnostic>>(e => [e])
```

Comment:
we could make it point-free with `(Array.of)` instead of `(e => [e])`

Reply:
@XVincentX, I must stay that's interesting. So how does `e => [e]` guarantee that? I'm just asking out of curiosity :+1: